### PR TITLE
Add an isEmpty property to Rect

### PR DESCRIPTION
**Pull request by abarth**
Also trace image decoding.

### CODE REVIEW

File: sky/engine/core/loader/CanvasImageDecoder.cpp [hunk 1]
```diff
@@ -51,6 +52,7 @@ void CanvasImageDecoder::OnDataAvailable(const void* data, size_t num_bytes) {
 }
 
 void CanvasImageDecoder::OnDataComplete() {
+  TRACE_EVENT0("blink", "CanvasImageDecoder::OnDataComplete");
```
Comment (krisgiesing) [inline sky/engine/core/loader/CanvasImageDecoder.cpp]: Did you mean to check this in as part of the same PR?

**Comment by krisgiesing** [comment]
LGTM

File: sky/engine/core/loader/CanvasImageDecoder.cpp (hunk 1 above)
Comment (abarth) [inline sky/engine/core/loader/CanvasImageDecoder.cpp]: I can split it up if you like, but they're both sort of trivial changes.

**Comment by krisgiesing** [comment]
Just checking that it was intended

**Pull request merged**